docs/source/native/class.rst

```diff
@@ -0,0 +1,159 @@
+Classes
+=======
+
+.. note::
+
+   This is by no means complete. Please add anything you know.
+
+Declaring Classes
+-----------------
+
+To declare a class, first add the ``untyped`` keyword and the class as a global variable at file level.
+
+.. code-block:: javascript
+
+    untyped
+    global var ExampleClass
+
+The ``untyped`` declaration is required because instances have an unknown type and it's not possible to use classes as types.
+
+``global var [classname]`` represents the class. After declaring the class inside of a function you can use it anywhere. It's not possible to declare classes on local variables.
+
+You can then add the declaration inside of a function.
+
+Most classes use a constructor. A constructor is a function of the instance that gets executed on object creation.
+
+.. code-block:: javascript
+
+    void function initClient() {
+        class ExampleClass {
+            constructor(){print("Instance of ExampleClass created");}
+        }
+    }
+
+You can require parameters in the constructor. Keep in mind that you have to pass those when creating an object.
+
+The types of the parameters have to be ``var``, but you can ``expect`` types in the constructor as usual.
+
+.. code-block:: javascript
+
+    class ExampleClass {
+            propertyString // Actual type is var
+            propertyInt // Actual type is var
+            constructor( var pString, var pInt ) {
+                this.propertyString = expect string(pString);
+                this.propertyInt = expect int(pInt); 
+            }
+    }
+
+    // See section "Declaring Objects" for more information on object creation
+    var obj = ExampleClass( "foo", 1 );
+    printt(obj.propertyString, obj.propertyString ) // foo, 1
+    var lObj = ExampleClass(); tObj = ExampleClass( "" , 0 , 0); // Both throw an error compile time because parameters don't match with the constructor
+
+Usually objects have properties. To define them, just add their identifier into the class without type declaration. The properties will be of type ``var``. However, you are required to set a default value of a property. This may be ``null``.
+
+Every object has a reference to itself called ``this``. You can change parameters of an object by reference.
+
+.. code-block:: javascript
+
+    void function initClient() {
+        class ExampleClass {
+            property = null
+            constructor( var parameter ) {
+                this.property = expect int(parameter);
+            }
+        }
+    }
+
+You can't use the class name as a type. Instead use ``var`` instead. You can't ``expect`` them either.
+
+Declaring Objects
```

Comment:
usually the term is "instantiating", declare, as i understand it, is defining what a name is, not setting its value

Reply:
"instantiating" is used to describe a class after an instance of that class has been created. "Initializing" would fit better because the variable (for example ExampleClass) gets initialized with a value (for example class ExampleClass{}).
However, the official Squirrel documentation refers to this as "Declaring Objects" so I just stuck with it.